server/src/main/java/com/objectcomputing/checkins/services/memberprofile/retentionreport/RetentionReportServicesImpl.java

```diff
@@ -0,0 +1,352 @@
+package com.objectcomputing.checkins.services.memberprofile.retentionreport;
+
+import com.objectcomputing.checkins.services.memberprofile.MemberProfile;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileServices;
+
+import javax.inject.Singleton;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+@Singleton
+public class RetentionReportServicesImpl implements RetentionReportServices {
+
+    private final MemberProfileServices memberProfileServices;
+
+    public RetentionReportServicesImpl(MemberProfileServices memberProfileServices) {
+        this.memberProfileServices = memberProfileServices;
+    }
+
+    @Override

```

Comment:
do we need any validations here like only admin can run the report ?

Reply:
Yes, we probably do. It wasn't in the AC, so I guess I missed it.

Reply:
Added validation